src/PowerManifold.jl

```diff
@@ -0,0 +1,364 @@
+            _read(rep_size, x, i),
+            FVector(TangentSpace, _read(rep_size, w.data, i)))
+    end
+    return v
+end
+
+function sharp!(M::PowerManifold, v::FVector{TangentSpaceType}, x, w::FVector{CotangentSpaceType})
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        sharp!(M.manifold,
+            FVector(TangentSpace, _write(rep_size, v.data, i)),
+            _read(rep_size, x, i),
+            FVector(CotangentSpace, _read(rep_size, w.data, i)))
+    end
+    return v
+end
+
+"""
+    is_manifold_point(M::ProductManifold, x; kwargs...)
+
+Check whether `x` is a valid point on the [`ProductManifold`](@ref) `M`.
+
+The tolerance for the last test can be set using the ´kwargs...`.
+"""
+function is_manifold_point(M::PowerManifold, x; kwargs...)
+    result = true
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        result &= is_manifold_point(M.manifold,
+            _read(rep_size, x, i); kwargs...)
+    end
+    return result
+end
+
+"""
+    is_tangent_vector(M::ProductManifold, x, v; kwargs... )
+
+Check whether `v` is a tangent vector to `x` on the [`ProductManifold`](@ref)
+`M`, i.e. atfer [`is_manifold_point`](@ref)`(M, x)`, and all projections to
+base manifolds must be respective tangent vectors.
+
+The tolerance for the last test can be set using the ´kwargs...`.
+"""
+function is_tangent_vector(M::PowerManifold, x, v; kwargs...)
+    result = true
+    rep_size = representation_size(M.manifold)
+    for i in get_iterator(M)
+        result &= is_tangent_vector(M.manifold,
+            _read(rep_size, x, i),
+            _read(rep_size, v, i); kwargs...)
+    end
+    return result
+end
+
+"""
+    PowerPointDistribution(M::PowerManifold, distribution)
+
+Power distribution on manifold `M`, based on `distribution`.
+"""
+struct PowerPointDistribution{TM<:PowerManifold, TD<:MPointDistribution, TX} <: MPointDistribution{TM}
+    manifold::TM
+    distribution::TD
+    x::TX
+end
+
+function support(d::PowerPointDistribution)
+    return MPointSupport(d.manifold)
+end
+
+function rand(rng::AbstractRNG, d::PowerPointDistribution)
+    x = similar(d.x)
+    _rand!(rng, d, x)
+    return x
+end
+
+function _rand!(rng::AbstractRNG, d::PowerPointDistribution, x::AbstractArray)
+    rep_size = representation_size(d.manifold.manifold)
+    for i in get_iterator(d.manifold)
+        _rand!(rng,
+            d.distribution,
+            _write(rep_size, x, i))
+    end
+    return x
+end
+
+"""
+    PowerFVectorDistribution([type::VectorBundleFibers], [x], distr)
+
+Generates a random vector at point `x` from vector space (a fiber of a tangent
+bundle) of type `type` using the power distribution of `distr`.
+
+Vector space type and `x` can be automatically inferred from distribution `distr`.
+"""
+struct PowerFVectorDistribution{TSpace<:VectorBundleFibers{<:VectorSpaceType, <:PowerManifold}, TD<:FVectorDistribution, TX} <: FVectorDistribution{TSpace, TX}
```

Comment:
What does the `F` in the name stand for? `Fiber`? Then one could also write that out.

Reply:
That's a reference to this type: https://github.com/JuliaNLSolvers/Manifolds.jl/blob/0ac128450df3e2501764a27b7955362dbd9925c8/src/VectorBundle.jl#L106 . Do you think it should be renamed to something like `FiberVector`?

Reply:
Ah, I see, so it's similar to `MPoint` and `TVector`? I didn't have that in mind. Then, of course, it'S fine to stay with that.